### DISCUSSION TITLE
[Broken] CROSS_PROCESS

### DISCUSSION THREAD

**Discussion by Struggleee** [SkyAPM/go2sky#144] [Q&A]
我在追蹤grpc service，
當我沒加sleep時會出現 [Broken] CROSS_PROCESS
``` golang
func sendRequest(client hello.HelloClient) {
	ctx := context.Background()
	exSpan, err := tracer.CreateExitSpan(ctx, "grpc-server", "grpc-server-domain:5051", func(headerKey, headerValue string) error {
		ctx = metadata.AppendToOutgoingContext(ctx, headerKey, headerValue)
		return nil
	})
	if err != nil {
		log.Fatalln(err)
	}
	exSpan.SetComponent(23)
	exSpan.SetSpanLayer(2)
	req := &hello.HelloRequest{
		Value: "test",
	}
	res, err := client.Hello(ctx, req)
	if err != nil {
		log.Fatalln(err)
	}
	exSpan.End()
	log.Println(res.Value)
}
```
![image](https://user-images.githubusercontent.com/33947281/147909381-440b328a-71d8-40ff-8d84-dbd33516a101.png)
![image](https://user-images.githubusercontent.com/33947281/147908918-1b8a51fa-1935-4ee8-8969-40182c670cd7.png)

當我加sleep 200ms 在 span.End()之後時，就正常運作。

``` golang
func sendRequest(client hello.HelloClient) {
	fmt.Println("Starting to do a RPC")
	ctx := context.Background()
	exSpan, err := tracer.CreateExitSpan(ctx, "grpc-server", "grpc-server-domain:5051", func(headerKey, headerValue string) error {
		ctx = metadata.AppendToOutgoingContext(ctx, headerKey, headerValue)
		return nil
	})
	if err != nil {
		log.Fatalln(err)
	}
	exSpan.SetComponent(23)
	exSpan.SetSpanLayer(2)
	req := &hello.HelloRequest{
		Value: "test",
	}
	res, err := client.Hello(ctx, req)
	if err != nil {
		log.Fatalln(err)
	}
	exSpan.End()
        time.Sleep(200 * time.Millisecond)
	log.Println(res.Value)
}
```
![image](https://user-images.githubusercontent.com/33947281/147909413-b14f864d-70a0-4a3a-8ff5-72b000d99512.png)
![image](https://user-images.githubusercontent.com/33947281/147909443-ed6d065a-7847-441b-847a-89716476b144.png)

請問這是我寫法有問題嗎？
謝謝。

**Comment by wu-sheng** [comment]
Please use English on GitHub. go2sky relies on your codes(go's context) to propagate the tracing context. And you should make sure all spans are closed properly. 

The ASF upstream doc provides explanation about what is VNode, https://skywalking.apache.org/docs/main/latest/en/faq/vnode/#what-is-vnode
